lambda/getLocation/index.js

```diff
@@ -1,202 +1,190 @@
 const AWS = require('aws-sdk');
 
+const Ajv = require("ajv")
+const ajv = new Ajv()
+
 const dynamo = new AWS.DynamoDB.DocumentClient();
+const DB = {
+    locations: "Locations",
+    parkingData: "ParkingLotData"
+}
+const querySchema = {
+    type: "object",
+    properties: {
+        CountyURL: { type: "string" },
+        ParkURL: { type: "string" }
+    },
+    additionalProperties: false
+}
+const validate = ajv.compile(querySchema)
 
-/**
- * Demonstrates a simple HTTP endpoint using API Gateway. You have full
- * access to the request and response payload, including headers and
- * status code.
- *
- * To scan a DynamoDB table, make a GET request with the TableName as a
- * query string parameter. To put, update, or delete an item, make a POST,
- * PUT, or DELETE request respectively, passing in the payload to the
- * DynamoDB API as a JSON body.
- */
- 
- const getParkingData = async () => {
-     let res;
-     try {
-          res = await dynamo.scan({ 
-                TableName: "ParkingLotData"
-            }).promise();
-     } catch (err) {
-         console.log(err)
-     }
-     if (res !== undefined) return res.Items;
- }
- 
-exports.handler = async (event, context) => {
 
-    let body;
-    let statusCode = '200';
-    const headers = {
-        'Content-Type': 'application/json',
-    };
-    
-    let parkingLotNameFilter = null
-    let recreationAreaFilter = null
-    let countyFilter = null
-    let parkUrlFilter = null
-    
-    if (event.queryStringParameters !== null && event.queryStringParameters !== undefined) {
-        parkingLotNameFilter = event.queryStringParameters.ParkingLotName
-        recreationAreaFilter = event.queryStringParameters.RecreationArea
-        countyFilter = event.queryStringParameters.CountyURL
-        parkUrlFilter = event.queryStringParameters.ParkURL
+// DynamoDB is not good at querying for non key values:
+// Should either switch to new DB or create new table with sort key by time
+// https://stackoverflow.com/questions/9297326/is-it-possible-to-order-results-with-query-or-scan-in-dynamodb
+const getLatestParkingData = async (lotID) => {
+    try {
+        const params = {
+            FilterExpression: "ParkingLotId = :lotID",
+            ExpressionAttributeValues: {
+                ':lotID': lotID
+            },
+            // ScanIndexForward: false, 
+            // limit: 1,
+            TableName: DB.parkingData
+        };
+        var data = await dynamo.scan(params).promise()
+        data = data.Items.sort((a, b) => b.LastUpdate - a.LastUpdate)
```

Comment:
This will be really bad as our number of parking data increases. Must find another solution like Using sort key on update time or better yet moving to relational DB